### PR TITLE
Added backend communication template

### LINKED ISSUE #64: Add backend communication template
Would be nice to have a template on how to do communication with the backend. I will propose something...

### PR DESCRIPTION
**Pull request by JonasHelming**
fixed #64

Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

How to test:

1. npm link
2. yo theia-extension => select "backend communication"
3. cd browser-app => yarn start
4. F1 => "Say hello on the backend (with a callback to the client) 
5. Check console to have printed two messages for both commands

### CODE REVIEW

File: src/app/index.ts [hunk 1]
```diff
@@ -130,6 +131,7 @@ module.exports = class TheiaExtension extends Base {
                     { value: ExtensionType.HelloWorld, name: 'Hello World' },
                     { value: ExtensionType.Widget, name: 'Widget' },
                     { value: ExtensionType.LabelProvider, name: 'LabelProvider' },
+                    { value: ExtensionType.Backend, name: 'Backend communication' },
```
Comment (vince-fugnitto) [inline src/app/index.ts:134]: minor: `Backend Communication` isntead of `Backend communication`

File: templates/backend/contribution.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    id: 'sayHelloOnBackend.command',
+    label: 'Say hello on the backend',
+};
+
+@injectable()
+export class <%= params.extensionPrefix %>CommandContribution implements CommandContribution {
+
+    constructor(
+        @inject(HelloBackendWithClientService) private readonly helloBackendWithClientService: HelloBackendWithClientService,
+        @inject(HelloBackendService) private readonly helloBackendService: HelloBackendService,
```
Comment (vince-fugnitto) [inline templates/backend/contribution.ts:20]: The injections should be outside the constructor, they align with our coding guidelines:
- https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#property-injection

File: templates/backend/contribution.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    registerCommands(registry: CommandRegistry): void {
+        registry.registerCommand(SayHelloViaBackendCommandWithCallBack, {
+            execute: () => this.helloBackendWithClientService.greet().then(r => console.log(r))
+        });
+        registry.registerCommand(SayHelloViaBackendCommand, {
+            execute: () => this.helloBackendService.sayHelloTo('World').then(r => console.log(r))
+        });
+    }
+}
```
Comment (vince-fugnitto) [inline templates/backend/contribution.ts:31]: minor: include a newline (same for all other files present in the pull-request).

File: templates/backend/contribution.ts (hunk 2 above)
Comment (JonasHelming) [reply templates/backend/contribution.ts:20]: Not consistent with other examples, too, suggest to fix it for all (I can do that):
https://github.com/theia-ide/generator-theia-extension/issues/66

File: templates/backend/contribution.ts (hunk 3 above)
Comment (JonasHelming) [reply templates/backend/contribution.ts]: done

File: src/app/index.ts (hunk 1 above)
Comment (JonasHelming) [reply src/app/index.ts]: done

**Comment by JonasHelming** [comment]
Can we merge this?

**Comment by vince-fugnitto** [comment]
> Can we merge this?

Sure, I assume you want to do this for the whole repo instead of fixing this pull-request?
- https://github.com/theia-ide/generator-theia-extension/pull/65#discussion_r443563028

**Comment by JonasHelming** [comment]
> > Can we merge this?
> 
> Sure, I assume you want to do this for the whole repo instead of fixing this pull-request?
> 
> * [#65 (comment)](https://github.com/theia-ide/generator-theia-extension/pull/65#discussion_r443563028)

yes...

**Comment by JonasHelming** [comment]
@marcdumais-work would you be able to perform a release?

**Comment by marcdumais-work** [comment]
> @marcdumais-work would you be able to perform a release?

Sure thing! Give me a few minutes.

**Comment by marcdumais-work** [comment]
`v0.1.19` is published.

**Comment by JonasHelming** [comment]
thanks!

**Pull request merged**